debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelope.java

```diff
@@ -36,7 +36,11 @@
 
     @Override
     public R apply(R r) {
-        SchemaBuilder valueSchemaBuilder = SchemaBuilder.struct();
+        String newValueSchemaName = r.valueSchema().name();
+        if (newValueSchemaName.endsWith(".Envelope")) {
```

Comment:
Why is it that you remove ".Envelope?"

Reply:
Because we are ripping out part of Envelope - so I think it make sense to name the schema after the collection only.